src/contracts/LiquidityProviderController.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import {OwnableOperable} from "./OwnableOperable.sol";
+import {ILiquidityProviderARM} from "./Interfaces.sol";
+
+/**
+ * @title Controller of ARM liquidity providers.
+ * @author Origin Protocol Inc
+ */
+contract LiquidityProviderController is Initializable, OwnableOperable {
+    /// @notice The address of the linked Application Redemption Manager (ARM).
+    address public immutable arm;
+
+    /// @notice The ARM's maximum allowed total assets.
+    uint256 public totalAssetsCap;
+    /// @notice The maximum allowed assets for each liquidity provider.
+    /// This is effectively a whitelist of liquidity providers as a zero amount prevents any deposits.
+    mapping(address liquidityProvider => uint256 cap) public liquidityProviderCaps;
```

Comment:
We need the boolean to disable / enable the per liquidityProvider caps.

Reply:
This code was changed when the `LiquidityProviderController` was moved out from the ARM contract. If there is no LiquidityProviderController configured then no liquidity caps will be checked.

```Solidity
    function _postDepositHook(uint256 assets) internal virtual override {
        if (liquidityProviderController != address(0)) {
            ILiquidityProviderController(liquidityProviderController).postDepositHook(msg.sender, assets);
        }
    }
```

Reply:
We still want to have the main cap on the total ETH amount, while allowing anyone to mint (regardless of their individual caps)